src/main/java/uk/co/jemos/podam/api/AbstractRandomDataProviderStrategy.java

```diff
@@ -542,8 +542,17 @@ public RandomDataProviderStrategy removeAttributeStrategy(
 	public AttributeStrategy<?> getStrategyForAnnotation(
 			final Class<? extends Annotation> annotationClass) {
 
-		return attributeStrategies.get(annotationClass);
+		if (attributeStrategies.containsKey(annotationClass)) {
+			return attributeStrategies.get(annotationClass);
+		} else {
+			for (Entry<Class<? extends Annotation>, AttributeStrategy<?>> attributeStrategy: attributeStrategies.entrySet()) {

```

Comment:
Here we iterate over all map. In practice probably all podam users have rather tiny maps, if not empty, still could be a problem for corner cases.

Reply:
changed